src/inspect_ai/model/_model.py

```diff
@@ -93,11 +95,32 @@
     set_active_generate_config,
 )
 from ._model_call import ModelCall
-from ._model_output import ModelOutput, ModelUsage
+from ._model_output import (
+    PerfectCacheUsage,
+    ModelOutput,
+    ModelUsage,
+)
 
 logger = logging.getLogger(__name__)
 
 
+class _TrieNode(dict):
+    __slots__ = ("tokens",)
+
+    def __init__(self):
+        super().__init__()
+        self.tokens: int | None = None
+
+
+def _sig(msg: ChatMessage) -> int:
+    """64-bit signature mapping role|name|content to uint64"""
```

Comment:
I'd maybe prefer a more specific name like _message_signature. Also might be good to note what this is used for.